File: js/examples/image_generation_example.js
const { doAPIRequest } = require('../aki_io');
const fs = require('fs');

// Configuration
const CONFIG = {
    ENDPOINT: 'z_image_turbo',
    API_KEY: 'fc3a8c50-b12b-4d6a-ba07-c9f6a6c32c37'
};


// Image generation parameters
const params = {
    prompt: 'Astronaut on Mars holding a banner which states "AKI is happy to serve your model" during sunset sitting on a giant yellow rubber duck',
    height: 1024,
    width: 1024,
    steps: 8,
    wait_for_result: true
};

console.log('🚀 Starting image generation...');

// Make the API request
doAPIRequest(
    CONFIG.ENDPOINT,
    CONFIG.API_KEY,
    params,
    (result) => {
        // Error handling
        if (result && result.success === false) {
            console.error('❌ API Error:', result.error || 'Unknown error');
            return;
        }

        const images = result.images || [];
        if (images.length === 0) {
            console.log('⚠️  No images were generated');
            return;
        }

        // Save the first image
        const imageData = images[0].includes(',') ? images[0].split(',')[1] : images[0];
        const outputPath = 'generated_image.png';
        fs.writeFileSync(outputPath, Buffer.from(imageData, 'base64'));
        
        console.log(`✅ Image saved as ${outputPath}`);
        console.log('✨ All done!');
    },
    (progress) => {
        const msg = progress.progress >= 0 
            ? `Progress: ${progress.progress}%` 
            : 'Starting...';
        process.stdout.write(`\r${msg}${' '.repeat(20)}`);
    }
);

File: python/examples/image_generation_example.py
import json
import base64
from pathlib import Path
from aki_io import Aki

def generate_image():
    
    # Define the image generation parameters
    params = {
        'prompt': 'Astronaut on Mars holding a banner which states "AKI is happy to serve your model" during sunset sitting on a giant yellow rubber duck',
        'seed': -1,
        'height': 512,
        'width': 512,
        'steps': 40,
        'provide_progress_images': 'none',
        'wait_for_result': True
    }

    # Call the AKI API
    result = do_aki_request(
        endpoint_name='qwen_image',
        api_key='fc3a8c50-b12b-4d6a-ba07-c9f6a6c32c37',
        params,
    )

    # Save the images
    images = result.get('images') or result.get('job_result', {}).get('images', [])
    if not images:
        print("No images returned by the API.")
        return result
    for i, img_b64 in enumerate(images):
        header, img_data = img_b64.split(',', 1) if ',' in img_b64 else (None, img_b64)
        img_bytes = base64.b64decode(img_data)
        filename = Path(__file__).parent / f'image_{i}.png'
        filename.write_bytes(img_bytes)
        print(f"Saved image to: {filename}")
    print(f"\nImage generation complete. {len(images)} image(s) saved.")
    return result

if __name__ == "__main__":
    generate_image()